### DISCUSSION TITLE
different behavior while accessing new and old style computed properties

### DISCUSSION THREAD

**Discussion by Briareos17** [livewire/livewire#7352] [7. Bugs (archived)]
### Livewire version

3.2.1

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. create methods:
`public function getComputedValueProperty() { return 1; }`
`
#[Computed]
public function anotherComputedValue() { return 2; }
`
2. Call both with snake_case and camelCase, first one will work fine on calling `$this->computed_value`, second one will fail `$this->another_computed_value`, this camelCase approach inconsistent with laravel's property paradigm

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by sueboy19** [comment]
Same Problem.

```php
#[Computed]
public function another_computed_value() {  }
```
view
```blade
{{ $this->another_computed_value() }}
```
Get Error !!

When I update 3.2.6 -> 3.3.0. At 3.2.6 No Error.
Now Code need $this->another_computed_value  -> $this->anotherComputedValue()

**Reply by joshhanley** [reply]
@sueboy19 you should not be using parens `()` with computed properties as you are just calling it like a function, which circumvents Livewire's computed property memoization and caching features (so you might as well just make it a normal method).

**Reply by sueboy19** [reply]
OK, let me try ~ Then fellback

**Reply by sueboy19** [reply]
I check my code. I mistake the demo. Now Put real project get problem.

```php
    #[Computed]
    public function voucher_detail_result()
    {
       ...
    }
```

```blade
@forelse ($this->voucher_detail_result as $key => $voucher_number_detail)
...
```

![image](https://github.com/livewire/livewire/assets/6514381/0388dc5f-46e2-4de0-95dd-a4e4f00abcad)

Still get error.

**Reply by joshhanley** [reply]
@sueboy19 the computed property method name needs to be camelCase as per PSR standards, but you can access the computed property with either camelCase or snake_case.

Like

```php
#[Computed]
public function voucherDetailResult()
{
   ...
}
```

```blade
@forelse ($this->voucher_detail_result as $key => $voucher_number_detail)

or

@forelse ($this->voucherDetailResult as $key => $voucher_number_detail)
```

Hope this helps!

**Reply by sueboy19** [reply]
Thank you. I change my project computed property method name to camelCase.
![未命名](https://github.com/livewire/livewire/assets/6514381/cf6c2482-ff0c-47d5-bab1-970595de2b80)

**Reply by joshhanley** [reply]
@sueboy19 no worries! 🙂
